### PR TITLE
Add the `WpBlock` custom element

### PR DESCRIPTION
**Pull request by michalczaplinski**
This PR is adding the `<wp-block>` custom element to the frontend of woocommerce. 

The [`<wp-block>`](https://github.com/WordPress/block-hydration-experiments/blob/8ebdddf255a155b365a0be75df39757900ad4930/src/gutenberg-packages/frontend.js#L27) was originally created in the BHE repo. This PR is bringing it to WBHE.

### CODE REVIEW

File: assets/js/base/utils/bhe-frontend.tsx [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+		blockTypes: Map< string, ReactElement >;
+	}
+}
+
+declare global {
+	// eslint-disable-next-line @typescript-eslint/no-namespace, @typescript-eslint/no-unused-vars
+	namespace JSX {
+		interface IntrinsicElements {
+			// eslint-disable-next-line @typescript-eslint/naming-convention
+			'gutenberg-inner-blocks': React.DetailedHTMLProps<
```
Comment (ockham) [inline assets/js/base/utils/bhe-frontend.tsx]: We should probably also update the "nomenclature" here, see https://github.com/woocommerce/woocommerce-blocks-hydration-experiments/pull/21#issuecomment-1195598808 🙂 

(tl;dr `gutenberg` ➡️ `wp`, or `wp-block` in other contexts)

Comment (ockham) [reply assets/js/base/utils/bhe-frontend.tsx]: I've taken the liberty of applying some nomenclature fixes in f7b465b. Feel free to revert if you don't like them! 😄

Comment (michalczaplinski) [reply assets/js/base/utils/bhe-frontend.tsx]: Thanks @ockham ! I was going to do the same thing but you beat me to it :)

**Comment by michalczaplinski** [comment]
There are some action failures but they're not consequential:

- generate ZIP
  - It's missing some credentials that are saved in a secret, but we don't have it in this repo. I've manually disabled it.
- automate assigning for review 
  - It also depends on some secret and I've also manually disabled it. We can just assign reviewers manually 🤷 🙂 
- e2e tests
  - Those failures are also present in the upstream `woocommerce-blocks` repo (e.g. in https://github.com/woocommerce/woocommerce-blocks/runs/7616910538?check_suite_focus=true), so it's not our fault. I've asked the Woo team about this, perhaps we should disable those tests in our fork.

**Pull request merged**